openedx/adg/admin_dashboard/performance/views.py

```diff
@@ -1,3 +1,151 @@
 """
 Views for `performance` app
 """
+import csv
+from datetime import date
+
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.contrib.auth.models import User
+from django.db.models import Prefetch, Q
+from django.http import HttpResponse
+from django.views.generic import ListView, View
+from opaque_keys.edx.keys import CourseKey
+from six import StringIO
+
+from common.djangoapps.student.models import CourseEnrollment
+from lms.djangoapps.courseware.courses import get_course_by_id
+from openedx.adg.admin_dashboard.constants import DEFAULT_PAGINATE_BY
+from openedx.adg.admin_dashboard.performance.helpers import get_course_grades_for_users, get_course_grades_report_data
+from openedx.adg.lms.applications.models import BusinessLine, MultilingualCourseGroup
+from openedx.adg.lms.applications.rules import is_adg_admin
+
+
+class CourseGradesView(UserPassesTestMixin, ListView, View):
+    """
+    View to list and download grades of a course.
+    """
+
+    DEFAULT_COURSE_GRADES_PAGINATE_BY = DEFAULT_PAGINATE_BY
+    course = None
+    template_name = 'adg/admin_dashboard/performance/course_grades.html'
+    http_method_names = ('get', 'post')
+
+    def test_func(self):
+        """
+        Checks user permissions to access course grades.
+        Authenticated and staff user will be able to access course grades if he/she is a
+        superuser | adg admin | bu admin and course is a bu prerequisite for the same bu.
+
+        Returns:
+            bool: True if user has all permissions, False otherwise
+        """
+        user = self.request.user
+        self.course = self._get_course()
+
+        if not (user.is_authenticated and user.is_staff):
+            return False
```

Comment:
Where are the users being redirected in case they are unauthenticated? And where are they being redirected in case they are not admin users?

Reply:
They are redirected to login if not authenticated, If not an admin then 403 is raised.